OpenNefia.Content/UI/Element/UiPageModel.cs

```diff
@@ -0,0 +1,100 @@
+﻿using OpenNefia.Content.UI.Element.List;
+using OpenNefia.Core.Audio;
+using OpenNefia.Core.Locale;
+using OpenNefia.Core.UI.Element;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace OpenNefia.Content.UI.Element
+{
+    public class UiPageModel<T> : UiElement
+    {
+        private UiText PageText;
+
+        private IEnumerable<T> PagedElements;
+        private int ItemsPerPage;
+        private int CurrentPage;
+        private UiWindow? Window = default!;
+        public int PageCount => (PagedElements.Count() / Math.Max(1, ItemsPerPage));
+
+        public delegate void PageChanged();
+        public event PageChanged? OnPageChanged;
+
+        public UiPageModel()
+        {
+            PagedElements = Enumerable.Empty<T>();
+            PageText = new UiText(UiFonts.WindowPage);
+        }
+
+        public void Initialize(IEnumerable<T> elements, int itemsPerPage = 15)
+        {
+            PagedElements = elements;
+            ItemsPerPage = itemsPerPage;
+            ChangePage(0);
+        }
+
+        public IEnumerable<T> GetCurrentElements()
+        {
+            var en = PagedElements.Skip(ItemsPerPage * CurrentPage);
+            return en.Take(ItemsPerPage);
+        }
+
+        public void PageForward()
+        {
+            if (PageCount > CurrentPage)
+            {
+                ChangePage(CurrentPage + 1);
+            }
+        }
+
+        public void PageBackward()
+        {
+            if (CurrentPage >= 1)
+            {
+                ChangePage(CurrentPage - 1);
+            }
+        }
+
+        public void ChangePage(int page)
+        {
+            Sounds.Play(Prototypes.Protos.Sound.Pop1);
```

Comment:
Nitpick: could we mute this if the list is being initialized? (it plays when entering the charamake menu for the first time)

Reply:
right, forgot about that. will change.

Reply:
changed in 76f6762